packages/core/src/browser/dialogs.ts

```diff
@@ -185,11 +189,17 @@ export abstract class AbstractDialog<T> extends BaseWidget {
 
     abstract get value(): T;
 
-    isValid(value: T): string {
+    /**
+     * Return a validation error.
+     *
+     * Only if it returns a string of 0-length then valid, i.e. even if it returns an empty string of non 0-length then invalid.
+     * It can be used to disable control elements without displaying an error.
+     */
+    protected isValid(value: T): string {
```

Comment:
Proposal: `protected isValid(value: T): true | { errorMessage: string }` or just `protected isValid(value: T): true | string`?
And then we could get rid of the documentation because you can guess it from the signature.
This current approach is the good old Java way. What do you think?

Reply:
Good idea :) I was refactoring to do it, but then realized that an empty string would do and reverted. I will add it.